engine/lib/phx/src/engine/main_loop.rs

```diff
@@ -255,9 +255,9 @@ impl ApplicationHandler for MainLoop {
         engine.input.update_gamepad(|state| state.update());
 
         // Let Lua script perform frame operations
-        engine.call_lua("AppFrame").unwrap_or_else(|e| {
-            panic!("Error calling AppInit: {}", e);
-        });
+        // Dispatch engine events
+        let lua = engine.lua.as_ref();
+        engine.event_bus.dispatch_all(lua);
```

Comment:
If we change `dispatch_all` to take a `&mut Lua`, then change this to 
```suggestion
        engine.event_bus.dispatch_all(lua.borrow_mut());
```

Reply:
lua ref does not have borrow_mut() and with use std::borrow::BorrowMut; it returns a Ref again xd

mismatched types
        expected struct `Ref<'_, _>`
found mutable reference `&mut Ref<'_, _>`

Reply:
I have to check it but I'm almost 100% sure there is a way to get `&mut` from `Ref`.

Reply:
Oh, that's probably because `engine.lua.as_ref()` is returning an `Ref<Lua>` already. It's because `engine.lua` is an `Rf<Lua>`

I'm pretty sure you need to run:
```
let lua = engine.lua.as_mut();
engine.event_bus.dispatch_all(&mut *lua)
```

Reply:
Although thinking about it, do we need an `&mut Lua` everywhere? I think `&Lua` is enough to call functions.